src/Idris/REPL.idr

```diff
@@ -607,6 +609,38 @@ loadMainFile f
            [] => pure (FileLoaded f)
            _ => pure (ErrorsBuildingFile f errs)
 
+docsOrSignature : {auto c : Ref Ctxt Defs} ->
+                  {auto s : Ref Syn SyntaxInfo} ->
+                  FC -> Name -> Core (List String)
+docsOrSignature fc n
+    = do syn  <- get Syn
+         defs <- get Ctxt
+         all@(_ :: _) <- lookupCtxtName n (gamma defs)
+             | _ => throw (UndefinedName fc n)
+         let ns@(_ :: _) = concatMap (\n => lookupName n (docstrings syn))
+                                     (map fst all)
+             | [] => typeSummary defs
+         getDocsFor fc n 
+  where
+    typeSummary : Defs -> Core (List String)
+    typeSummary defs = do Just def <- lookupCtxtExact n (gamma defs)
+                            | Nothing => pure []
+                          ty <- normaliseHoles defs [] (type def)
+                          pure [(show n) ++ " : " ++ (show !(resugar [] ty))]
+
+equivTypes : {auto c : Ref Ctxt Defs} ->
+             (ty1 : ClosedTerm) ->
+             (ty2 : ClosedTerm) ->
+             Core Bool
+equivTypes ty1 ty2 = do defs <- get Ctxt
+                        True <- pure (!(getArity defs [] ty1) == !(getArity defs [] ty2))
+                          | False => pure False
```

Comment:
I think this is `when`

```suggestion
                        when (!(getArity defs [] ty1) == !(getArity defs [] ty2))) (pure False)
```

Reply:
I don't think that captures the same meaning. I want the entire `do` block to short circuit with `pure False` when the arities differ as opposed to executing something conditionally (unless you're talking about the entire rest of the block as the thing being optionally executed).

That said, it would not surprise me if there _was_ something better suited to the task here. It feels awkward to take the `False` case and just pass it along to `pure False` but that happens to be how I am getting the short circuit behavior, too.

Reply:
Ah yes you're right! This is the right thing to do then :)